### PR TITLE
feat: Add workaround/fix for rendering issue in MSFS checklists

### PR DESCRIPTION
**Pull request by Chryssie**
## Summary of Changes

Adds a separate package that fixes a bug for rendering the Asobo checklists with nested blocks.

## Screenshots (if necessary)

Without the fix nested `Block` elements in checklists are rendered incorrectly (the alignment is wrong) and contain phantom (duplicate) items:
broken nested blocks example

The fix improves the indentation and prevents the phantom items from being generated.
working nested blocks example

## Additional context

This was originally a part of #7627, however, I have split it out to this pull request.

⚠ This change does impact the entire game and other aircraft as it overrides (in the virtual file system, no files on disk are changed) some core game files.

However, it has now been updated so that the override in it's own package that can be uninstalled (or not installed in the first place) in the unlikely event that it causes a conflict with another addon.

Discord username (if different from GitHub): Chryssie#1984

## Testing instructions

There should be no impact from this change by itself as the checklists do not currently use nested blocks. It needs to be tested in conjunction with #7627.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -65,6 +65,9 @@
 /fbw-common/msfs-avionics-mirror/src/sdk/build
 /fbw-common/msfs-avionics-mirror/src/sdk/build/**
 
+!/fbw-msfs-checklist-fix/**
+/fbw-msfs-checklist-fix/out
```
Comment (frankkopp) [inline .gitignore:69]: Can you add - this makes sure the structure within these folders is identical:

/fbw-msfs-checklist-fix/*

!/fbw-msfs-checklist-fix/README.md

!/fbw-msfs-checklist-fix/docs/

!/fbw-msfs-checklist-fix/docs/**

!/fbw-msfs-checklist-fix/src/

!/fbw-msfs-checklist-fix/src/**

/fbw-msfs-checklist-fix/out

Also can't hurt to write a README.md to describe what this folder is for.

**Review by frankkopp — CHANGES_REQUESTED** [review]
Can you take a look at the gitignore? the rest is good.

File: .github/workflows/msfs-checklist-fix.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths:
+      - 'fbw-msfs-checklist-fix/**'
+      - 'scripts/build_msfs_checklist_fix.js'
+      - 'scripts/build_msfs_checklist_fix.sh'
+      - 'igniter.config.mjs'
+      - 'package.json'
+      - 'package-lock.json'
```
Comment (frankkopp) [inline .github/workflows/msfs-checklist-fix.yml:15]:
```suggestion
```

Remove this

**Review by frankkopp — CHANGES_REQUESTED** [review]

**Review by frankkopp — APPROVED** [review]

**Review by Benjozork — CHANGES_REQUESTED** [review]
The output package name does not follow the convention set by the MSFS documentation - this convention is useful for understanding what a package does.

image

I suggesst `flybywire-ingamepanels-checklist-fix`. The official package name, for reference, is `fs-base-ingamepanels-checklist`.

**Comment by Chryssie** [comment]
Thanks @Benjozork, I've updated the package name (along with all of the folders) to `flybywire-ingamepanels-checklist-fix` (except the root folder which is `fbw-ingamepanels-checklist-fix` in line with the other packages).

**Review by Benjozork — APPROVED** [review]

**Pull request merged**